### ISSUE TITLE
Incorrect "bindings never used" for destructuring struct patterns in functions

### ISSUE THREAD

**Issue by afetisov** [intellij-rust/intellij-rust#7742]
## Environment

* **IntelliJ Rust plugin version:** 0.4.155.4075-212-nightly
* **Rust toolchain version:** 1.54.0 (a178d0322 2021-07-26) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.2 (CL-212.4746.93)
* **Operating system:** Linux 5.4.0-77-generic
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

The following code produces `Binding is never used` warning even though the bindings are used in the function body. At the same time the reference search works correctly, and selecting one of the idents `x` correctly highlights the other. 

```rust
pub struct Foo { a: u32, b: u32 }
pub fn f(Foo { a: x, b: y }: Foo) -> u32 {
               // ^ Binding `x` is never used
                     // ^ Binding `y` is never used
    x + y       
}
```

The warning doesn't trigger if we do not rename the struct fields:
```rust
pub struct Foo { a: u32, b: u32 }
pub fn f(Foo { a, b }: Foo) -> u32 {
    // no warnings
    a + b
}
```

Nor does it fire if we use a destructuring `let` in the function body:
```rust
pub struct Foo { a: u32, b: u32 }
pub fn f(t: Foo) -> u32 {
    let Foo { a: x, b: y } = t;
    // no warnings
    x + y
}
```
